---
title: "ENS in inflammation: DGE & GSEA"
date: "`r Sys.Date()`"
output:
      html_document:
        toc: true
        toc_float: true
        number_sections: true
        collapsed: true
        fig_caption: yes
        df_print: paged
---

DGE and Gene Ontology Analysis of the Enteric Nervous System during Inflammation

Muscularis of infected and naive Snap25CreINTACT mice was isolated. Nuclei isolation was performed and neuronal nuclei were sort-purified (GFP+). Bulk RNA-Sequencing was done and the results are shownbelow.

```{r setup, include=FALSE, message = FALSE}
knitr::opts_chunk$set(echo = TRUE)
# knitr::opts_knit$set(root.dir = "/Volumes/PortableSSD/1_PhD/1_Projects/1_Neurons/2_SeqAnalysis/1_Mine")

# load importante libraries

library("DESeq2")
library("limma")
library("gProfileR")
library("httr")
library("RJSONIO")
library(pheatmap)
library(dbplyr)
library(dplyr)
library(ggplot2)
library(tidyverse)
library(hrbrthemes)
library(viridis)
library(plotly)
library(RColorBrewer)
library(viridis)
library(readxl)
#library(DEGreport)
library(htmltools)
library(vctrs)
library(EnhancedVolcano)
library(readr)
library("topGO")
library("org.Mm.eg.db")
library("AnnotationDbi")
library("DOSE")
library("GO.db")
library("GSEABase")
library("clusterProfiler")
library("tidyr")
library("stringr")
library("rWikiPathways")
library("RCy3")
library(ReactomePA)
library(scales)
library(gridExtra)
library(grid)
library(ggrepel)
library(repr)
library(ggpubr)
library(knitr)
library(kableExtra)
library(DT)
library(xfun)
library(rmarkdown)
library(patchwork)
library(readr)
library(gprofiler2)
library(ggplot2)
library(gt)
library(gtExtras)



```

```{r Import data, echo=FALSE,warning=FALSE,message = FALSE}

### counts
#
# counts.ens <- as.data.frame(read_excel("1_ENS/Counts/ENS_Counts.xlsx"))
# counts.drg <- as.data.frame(read_excel("DRG/Counts/DRG_Counts.xlsx"))
# counts.smg <- as.data.frame(read_excel("SMG/Counts/SMG_Counts.xlsx"))
# counts.vg <- as.data.frame(read_excel("VG/Counts/VG_Counts.xlsx"))


#### norm counts

Neurons_NormCounts_symbol <- read.csv("../data/processed/Neurons_NormCounts_symbol.csv")
nc <- as.data.frame(Neurons_NormCounts_symbol)
nc <- unique(nc)
nc <- nc[!duplicated(nc$gene_name),]
nc <- na.omit(nc)
rownames(nc) <- nc$gene_name


### DESeq2

ens.nippo <- read_excel("../data/processed/ENS_DESeq2_Nippo.xlsx")
ens.dss <- read_excel("../data/processed/ENS_DESeq2_DSS.xlsx")
ens.th17 <- read_excel("../data/processed/ENS_DESeq2_Th17.xlsx")

#
# drg.nippo <- read_excel("2_DRG/DESeq2/DRG_DESeq2_Nippo.xlsx")
# drg.dss <- read_excel("2_DRG/DESeq2/DRG_DESeq2_DSS.xlsx")
# drg.th17 <- read_excel("2_DRG/DESeq2/DRG_DESeq2_Th17.xlsx")
#
#
# smg.nippo <- read_excel("3_SMG/DESeq2/SMG_DESeq2_Nippo.xlsx")
# smg.dss <- read_excel("3_SMG/DESeq2/SMG_DESeq2_DSS.xlsx")
# smg.th17 <- read_excel("3_SMG/DESeq2/SMG_DESeq2_Th17.xlsx")
#
#
# vg.nippo <- read_excel("4_VG/DESeq2/VG_DESeq2_Nippo.xlsx")
# vg.dss <- read_excel("4_VG/DESeq2/VG_DESeq2_DSS.xlsx")
# vg.th17 <- read_excel("4_VG/DESeq2/VG_DESeq2_Th17.xlsx")

## for Volcano

ens.nippo.v <- read.delim("../data/raw/DESeq2_ENS_Nippo_results.tabular", header=F)
ens.dss.v <- read.delim("../data/raw/DESeq2_ENS_DSS_results.tabular", header=F)
ens.th17.v <- read.delim("../data/raw/DESeq2_ENS_Th17_results.tabular", header=F)


# drg.nippo.v <- read.delim("2_DRG/Galaxy/DESeq2_DRG_Nippo_results.tabular", header=F)
# drg.dss.v <- read.delim("2_DRG/Galaxy/DESeq2_DRG_DSS_results.tabular", header=F)
# drg.th17.v <- read.delim("2_DRG/Galaxy/DESeq2_DRG_Th17_results.tabular", header=F)
#
#
# smg.nippo.v <- read.delim("3_SMG/Galaxy/DESeq2_SMG_Nippo_results.tabular", header=F)
# smg.dss.v <- read.delim("3_SMG/Galaxy/DESeq2_SMG_DSS_results.tabular", header=F)
# smg.th17.v <- read.delim("3_SMG/Galaxy/DESeq2_SMG_Th17_results.tabular", header=F)
#
#
# vg.nippo.v <- read.delim("4_VG/Galaxy/DESeq2_VG_Nippo_results.tabular", header=F)
# vg.dss.v <- read.delim("4_VG/Galaxy/DESeq2_VG_DSS_results.tabular", header=F)
# vg.th17.v <- read.delim("4_VG/Galaxy/DESeq2_VG_Th17_results.tabular", header=F)

colnames(ens.nippo.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
colnames(ens.dss.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
colnames(ens.th17.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")


# colnames(drg.nippo.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
# colnames(drg.dss.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
# colnames(drg.th17.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
#
#
# colnames(smg.nippo.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
# colnames(smg.dss.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
# colnames(smg.th17.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
#
#
# colnames(vg.nippo.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
# colnames(vg.dss.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")
# colnames(vg.th17.v) <- c("GeneID","BaseMEan","Log2FC","StdErr","WaldStat","pval","padj")



anno_df <- data.frame(GeneID = ens.nippo.v$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
ens.nippo.v <- merge(ens.nippo.v, anno_df, by = "GeneID", all=TRUE)

anno_df <- data.frame(GeneID = ens.dss.v$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
ens.dss.v <- merge(ens.dss.v, anno_df, by = "GeneID", all=TRUE)

anno_df <- data.frame(GeneID = ens.th17.v$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
ens.th17.v <- merge(ens.th17.v, anno_df, by = "GeneID", all=TRUE)



anno_df <- data.frame(GeneID = nc$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
nc <- merge(nc, anno_df, by = "GeneID", all=TRUE)


```

# DESeq2

## Results {.tabset}
### Nipoo

```{r DGESeq2 Results Nippo, echo=FALSE, paged.print=TRUE, results='asis',warning=FALSE}


ens.nippo <- na.omit(ens.nippo)
ens.nippo <- transform(ens.nippo, pvalue=sapply(pvalue, formatC, format = "e", digits=3))
ens.nippo <- transform(ens.nippo, padj=sapply(padj, formatC, format = "e", digits=3))
ens.nippo <-ens.nippo  %>% mutate_at(vars("BaseMean", "Log2FC", "StdErr", "Wald-Stats"), funs(round(., 3)))


ens.nippo %>%
  gt() %>%
  tab_header(
    title = 'DGE: ENS Nippo vs Naive',
    subtitle = 'DESeq2 Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### DSS

```{r DGESeq2 Results DSS, echo=FALSE, paged.print=TRUE, results='asis',warning=FALSE}


ens.dss <- na.omit(ens.dss)
ens.dss <- transform(ens.dss, pvalue=sapply(pvalue, formatC, format = "e", digits=3))
ens.dss <- transform(ens.dss, padj=sapply(padj, formatC, format = "e", digits=3))
ens.dss <-ens.dss  %>% mutate_at(vars("BaseMean", "Log2FC", "StdErr", "Wald-Stats"), funs(round(., 3)))


ens.dss %>%
  gt() %>%
  tab_header(
    title = 'DGE: ENS DSS vs Naive',
    subtitle = 'DESeq2 Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### Th17

```{r DGESeq2 Results th17, echo=FALSE, paged.print=TRUE, results='asis',warning=FALSE}


ens.th17 <- na.omit(ens.th17)
ens.th17 <- transform(ens.th17, pvalue=sapply(pvalue, formatC, format = "e", digits=3))
ens.th17 <- transform(ens.th17, padj=sapply(padj, formatC, format = "e", digits=3))
ens.th17 <-ens.th17  %>% mutate_at(vars("BaseMean", "Log2FC", "StdErr", "Wald-Stats"), funs(round(., 3)))


ens.th17 %>%
  gt() %>%
  tab_header(
    title = 'DGE: ENS Th17 vs Naive',
    subtitle = 'DESeq2 Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```


<br>

## Volcano Plots {.tabset}

```{r Volcano Plots, echo=FALSE, warning=FALSE, message = FALSE}

######## nippo

ens.nippo.v <- na.omit(ens.nippo.v)

top_up <- ens.nippo.v[ens.nippo.v$Log2FC > 1.5 & ens.nippo.v$padj < 0.05,]
top_up <- top_up[order(-top_up$Log2FC),]

top_down <- ens.nippo.v[ens.nippo.v$Log2FC < -1.5 & ens.nippo.v$padj < 0.05,]
top_down <- top_down[order(top_down$Log2FC),]


top <- c(top_up[1:50,"GeneName"],top_down[1:50,"GeneName"])


V.nippo <- EnhancedVolcano(ens.nippo.v,
                      lab = ens.nippo.v$GeneName,
                      selectLab = c(top),
                      #boxedLabels = TRUE,
                      #parseLabels = TRUE,
                      drawConnectors = TRUE,
                      widthConnectors = 0.25,
                      colConnectors = c("black"),
                      x = 'Log2FC',
                      y = 'pval',
                      title = 'ENS: Nippo vs Naive',
                      xlim = c(-10,10),
                      ylim = c(0,60),
                      pCutoff = 0.05,
                      FCcutoff = 0.5,
                      pointSize = 1.5,
                      labSize = 3.5,
                      col = c('#3e3e40', '#3e3e40', '#3e3e40',"#0066a1"),
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)


######## DSS

ens.dss.v <- na.omit(ens.dss.v)

top_up <- ens.dss.v[ens.dss.v$Log2FC > 1.5 & ens.dss.v$padj < 0.05,]
top_up <- top_up[order(-top_up$Log2FC),]

top_down <- ens.dss.v[ens.dss.v$Log2FC < -1.5 & ens.dss.v$padj < 0.05,]
top_down <- top_down[order(top_down$Log2FC),]


top <- c(top_up[1:50,"GeneName"],top_down[1:50,"GeneName"])


V.dss <- EnhancedVolcano(ens.dss.v,
                      lab = ens.dss.v$GeneName,
                      selectLab = c(top),
                      #boxedLabels = TRUE,
                      #parseLabels = TRUE,
                      drawConnectors = TRUE,
                      widthConnectors = 0.25,
                      colConnectors = c("black"),
                      x = 'Log2FC',
                      y = 'pval',
                      title = 'ENS: DSS vs Naive',
                      xlim = c(-10,10),
                      ylim = c(0,60),
                      pCutoff = 0.05,
                      FCcutoff = 0.5,
                      pointSize = 1.5,
                      labSize = 3.5,
                      col = c('#3e3e40', '#3e3e40', '#3e3e40',"red"),
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)


######## Th17

ens.th17.v <- na.omit(ens.th17.v)

top_up <- ens.th17.v[ens.th17.v$Log2FC > 1.5 & ens.th17.v$padj < 0.05,]
top_up <- top_up[order(-top_up$Log2FC),]

top_down <- ens.th17.v[ens.th17.v$Log2FC < -1.5 & ens.th17.v$padj < 0.05,]
top_down <- top_down[order(top_down$Log2FC),]


top <- c(top_up[1:50,"GeneName"],top_down[1:50,"GeneName"])


V.th17 <- EnhancedVolcano(ens.th17.v,
                      lab = ens.th17.v$GeneName,
                      selectLab = c(top),
                      #boxedLabels = TRUE,
                      #parseLabels = TRUE,
                      drawConnectors = TRUE,
                      widthConnectors = 0.25,
                      colConnectors = c("black"),
                      x = 'Log2FC',
                      y = 'pval',
                      title = 'ENS: Th17 vs Naive',
                      xlim = c(-10,10),
                      ylim = c(0,60),
                      pCutoff = 0.05,
                      FCcutoff = 0.5,
                      pointSize = 1.5,
                      labSize = 3.5,
                      col = c('#3e3e40', '#3e3e40', '#3e3e40',"darkgreen"),
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)


```

```{r enhancedVolcano plot, message = FALSE, echo=FALSE, warning=FALSE, fig.height=12,fig.width=25, fig.fullwidth=TRUE}

ggarrange(V.nippo, V.dss, V.th17, ncol=3, nrow=1)

```
<br><br><br><br>

Gene Lists are filtered for Log2FoldChange: upregulated \> 1.5 & downregulated \< -1.5 and padj \< 0.05 For Analysis, gene symbols were used, and ordered query was performed

# GSEA

## gProfiler Enrichment Analysis {.tabset}

```{r GSEA with  gProfiler, message = FALSE, echo=FALSE, warning=FALSE, results='hide'}

## ENS Nippo

ens.nippo <- ens.nippo[ens.nippo$BaseMean > 30, ]

up.genes <- ens.nippo[ens.nippo$Log2FC > 1.5, ]
up.genes.ordered <- up.genes[order(up.genes$Log2FC, decreasing = TRUE),]

dn.genes <- ens.nippo[ens.nippo$Log2FC < -1.5, ]
dn.genes.ordered <- dn.genes[order(up.genes$Log2FC, decreasing = FALSE),]

multi_gp_ens_nippo = gost(list("up-regulated" = up.genes.ordered$gene_name,
                     "down-regulated" = dn.genes.ordered$gene_name),organism = "mmusculus", ordered_query = TRUE)

gp_ens_nippo <- gostplot(multi_gp_ens_nippo, interactive = TRUE)

## DSS

ens.dss <- ens.dss[ens.dss$BaseMean > 30, ]

up.genes <- ens.dss[ens.dss$Log2FC > 1.5, ]
up.genes.ordered <- up.genes[order(up.genes$Log2FC, decreasing = TRUE),]

dn.genes <- ens.dss[ens.dss$Log2FC < -1.5, ]
dn.genes.ordered <- dn.genes[order(up.genes$Log2FC, decreasing = FALSE),]


multi_gp_ens_dss = gost(list("up-regulated" = up.genes.ordered$gene_name,
                     "down-regulated" = dn.genes.ordered$gene_name),organism = "mmusculus", ordered_query = TRUE)



gp_ens_dss <- gostplot(multi_gp_ens_dss, interactive = TRUE)

## Th17

ens.th17 <- ens.th17[ens.th17$BaseMean > 30, ]

up.genes <- ens.th17[ens.th17$Log2FC > 1.5, ]
up.genes.ordered <- up.genes[order(up.genes$Log2FC, decreasing = TRUE),]

dn.genes <- ens.th17[ens.th17$Log2FC < -1.5, ]
dn.genes.ordered <- dn.genes[order(up.genes$Log2FC, decreasing = FALSE),]


multi_gp_ens_th17 = gost(list("up-regulated" = up.genes.ordered$gene_name,
                     "down-regulated" = dn.genes.ordered$gene_name),organism = "mmusculus", ordered_query = TRUE)



gp_ens_th17 <- gostplot(multi_gp_ens_th17, interactive = TRUE)


```

### Nippo

```{r gProfiler nippo plot, message = FALSE, echo=FALSE, warning=FALSE}

ggplotly(gp_ens_nippo)

```

<br>

```{r gProfiler nippo, message = FALSE, echo=FALSE, warning=FALSE}


gp_ens_nippo_table <- multi_gp_ens_nippo[["result"]]
gp_ens_nippo_table <- transform(gp_ens_nippo_table, p_value=sapply(p_value, formatC, format = "e", digits=3))
gp_ens_nippo_table <- gp_ens_nippo_table %>% mutate_at(vars("precision","recall"), funs(round(., 3)))
gp_ens_nippo_table <- subset(gp_ens_nippo_table, select=-c(2))
table_ens_nippo <- gp_ens_nippo_table[,c("term_id","term_name","source","query","p_value","term_size","query_size","intersection_size","precision","effective_domain_size","parents")]

table_ens_nippo %>%
  gt() %>%
  tab_header(
    title = 'ENS Nippo vs Naive',
    subtitle = 'gProfiler Analysis'
  ) %>%
  cols_label(
   term_id = "Term ID",
   term_name = "Term Name",
   source = "Source",
   query = "Up/Down",
   p_value = "P-Value",
   term_size = "Term Size",
   query_size = "Query Size",
   intersection_size = "Intersection Size",
   precision = "Precision",
   effective_domain_size = "Effective Domain Size",
   parents = "Parents"
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )



```

<br><br>

### DSS

```{r gProfiler dss plot, message = FALSE, echo=FALSE, warning=FALSE}

ggplotly(gp_ens_dss)

```

<br>

```{r gProfiler dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

gp_ens_dss_table <- multi_gp_ens_dss[["result"]]
gp_ens_dss_table <- transform(gp_ens_dss_table, p_value=sapply(p_value, formatC, format = "e", digits=3))
gp_ens_dss_table <- gp_ens_dss_table %>% mutate_at(vars("precision","recall"), funs(round(., 3)))
gp_ens_dss_table <- subset(gp_ens_dss_table, select=-c(2))
table_ens_dss <- gp_ens_dss_table[,c("term_id","term_name","source","query","p_value","term_size","query_size","intersection_size","precision","effective_domain_size","parents")]

table_ens_dss %>%
  gt() %>%
  tab_header(
    title = 'ENS DSS vs Naive',
    subtitle = 'gProfiler Analysis'
  ) %>%
  cols_label(
   term_id = "Term ID",
   term_name = "Term Name",
   source = "Source",
   query = "Up/Down",
   p_value = "P-Value",
   term_size = "Term Size",
   query_size = "Query Size",
   intersection_size = "Intersection Size",
   precision = "Precision",
   effective_domain_size = "Effective Domain Size",
   parents = "Parents"
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )


```

<br><br>


### Th17

```{r gProfiler Th17 plot, message = FALSE, echo=FALSE, warning=FALSE}

ggplotly(gp_ens_th17)

```

<br>

```{r Result th17 , echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

gp_ens_th17_table <- multi_gp_ens_th17[["result"]]
gp_ens_th17_table <- transform(gp_ens_th17_table, p_value=sapply(p_value, formatC, format = "e", digits=3))
gp_ens_th17_table <- gp_ens_th17_table %>% mutate_at(vars("precision","recall"), funs(round(., 3)))
gp_ens_th17_table <- subset(gp_ens_th17_table, select=-c(2))
table_ens_th17 <- gp_ens_th17_table[,c("term_id","term_name","source","query","p_value","term_size","query_size","intersection_size","precision","effective_domain_size","parents")]

table_ens_th17 %>%
  gt() %>%
  tab_header(
    title = 'ENS Th17 vs Naive',
    subtitle = 'gProfiler Analysis'
  ) %>%
  cols_label(
   term_id = "Term ID",
   term_name = "Term Name",
   source = "Source",
   query = "Up/Down",
   p_value = "P-Value",
   term_size = "Term Size",
   query_size = "Query Size",
   intersection_size = "Intersection Size",
   precision = "Precision",
   effective_domain_size = "Effective Domain Size",
   parents = "Parents"
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )



```


<br> <br> <br>

## clusterProfiler Enrichment Analysis {.tabset}

```{r GSEA with cluster Profiler, message = FALSE, echo=FALSE, warning=FALSE, results='hide'}

###### ENS #################################


ens.nippo <- ens.nippo[ens.nippo$padj <= 0.05, ]
ens.nippo2 <- ens.nippo[abs(ens.nippo$Log2FC) > 2,]
ens.nippo2 <- ens.nippo2[order(ens.nippo2$padj),]

genes.entrez <- bitr(ens.nippo2$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)

## gene list for upregulated genes
up.genes <- ens.nippo[ens.nippo$Log2FC > 1.5, ]
up.genes.entrez <- bitr(up.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## gene list for downregulated genes
dn.genes <- ens.nippo[ens.nippo$Log2FC < -1.5, ]
dn.genes.entrez <- bitr(dn.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## backbone gene list (all genes)
bkgd.genes <- Neurons_NormCounts_symbol$GeneID
bkgd.genes.entrez <- bitr(bkgd.genes,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


##  https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/  #####

ego_up_ens_nippo <- clusterProfiler::enrichGO(
  gene     = up.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


ego_dn_ens_nippo <- clusterProfiler::enrichGO(
  gene     = dn.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)

ego_ens_nippo <- clusterProfiler::enrichGO(
  gene     = genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)

GO_up_ens_nippo <- as.data.frame(ego_up_ens_nippo)
GO_up_ens_nippo$query <- c("up-regulated")

GO_down_ens_nippo <- as.data.frame(ego_dn_ens_nippo)
GO_down_ens_nippo$query <- c("down-regulated")



### DSS ###


ens.dss <- ens.dss[ens.dss$padj <= 0.05, ]
ens.dss2 <- ens.dss[abs(ens.dss$Log2FC) > 2,]
ens.dss2 <- ens.dss2[order(ens.dss2$padj),]

#gene_list <- gene_list[abs(gene_list) > 1.5]
genes.entrez <- bitr(ens.dss2$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


## gene list for upregulated genes
up.genes <- ens.dss[ens.dss$Log2FC > 1.5, ]
up.genes.entrez <- bitr(up.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## gene list for downregulated genes
dn.genes <- ens.dss[ens.dss$Log2FC < -1.5, ]
dn.genes.entrez <- bitr(dn.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


##  https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/  #####

ego_up_ens_dss <- clusterProfiler::enrichGO(
  gene     = up.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)

ego_dn_ens_dss <- clusterProfiler::enrichGO(
  gene     = dn.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)

ego_ens_dss <- clusterProfiler::enrichGO(
  gene     = genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


GO_up_ens_dss <- as.data.frame(ego_up_ens_dss)
GO_up_ens_dss$query <- c("up-regulated")

GO_down_ens_dss <- as.data.frame(ego_dn_ens_dss)
GO_down_ens_dss$query <- c("down-regulated")

GO_ens_dss <- as.data.frame(ego_ens_dss)
GO_ens_dss$query <- c("down-regulated")


### Th17 ###

ens.th17 <- ens.th17[ens.th17$padj <= 0.05, ]
ens.th172 <- ens.th17[abs(ens.th17$Log2FC) > 2,]
ens.th172 <- ens.th172[order(ens.th172$padj),]

#gene_list <- gene_list[abs(gene_list) > 1.5]
genes.entrez <- bitr(ens.th172$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)

## gene list for upregulated genes
up.genes <- ens.th17[ens.th17$Log2FC > 1.5, ]
up.genes.entrez <- bitr(up.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## gene list for downregulated genes
dn.genes <- ens.th17[ens.th17$Log2FC < -1.5, ]
dn.genes.entrez <- bitr(dn.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


##  https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/  #####

ego_up_ens_th17 <- clusterProfiler::enrichGO(
  gene     = up.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


ego_dn_ens_th17 <- clusterProfiler::enrichGO(
  gene     = dn.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)

ego_ens_th17 <- clusterProfiler::enrichGO(
  gene     = genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


GO_up_ens_th17 <- as.data.frame(ego_up_ens_th17)
GO_up_ens_th17$query <- c("up-regulated")

GO_down_ens_th17 <- as.data.frame(ego_dn_ens_th17)
GO_down_ens_th17$query <- c("down-regulated")

```

```{r GSEA with cluster Profiler enrichGO, message = FALSE, echo=FALSE, warning=FALSE}


gg_up_ens_nippo <- dotplot(ego_up_ens_nippo, showCategory=10) + ggtitle("ENS Nippo - UP") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))
gg_dn_ens_nippo <- dotplot(ego_dn_ens_nippo, showCategory=10) + ggtitle("ENS Nippo - DOWN") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))

gg_up_ens_dss <- dotplot(ego_up_ens_dss, showCategory=10) + ggtitle("ENS DSS - UP") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))
gg_dn_ens_dss <- dotplot(ego_dn_ens_dss, showCategory=10) + ggtitle("ENS DSS - DOWN") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))

gg_up_ens_th17 <- dotplot(ego_up_ens_th17, showCategory=10) + ggtitle("ENS Th17 - UP") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))
gg_dn_ens_th17 <- dotplot(ego_dn_ens_th17, showCategory=10) + ggtitle("ENS Th17 - DOWN") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))

# legend <-  theme(legend.key.size = unit(0.2, 'cm'), #change legend key size
#                  legend.key.height = unit(0.5, 'cm'), #change legend key height
#                  legend.key.width = unit(0.3, 'cm'), #change legend key width
#                  legend.title = element_text(size=8), #change legend title font size
#                  legend.text = element_text(size=8)) #change legend text font size
#
# labels <- scale_y_discrete(labels = function(x) str_wrap(x, width = 200))
# axis <-  theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=10), axis.title.x=element_text(size=8),axis.title.y=element_text(size=0),legend.text = element_text(size = 2))
#
#
# ### make barplots from GO_BP top 10
#
# gg_up_ens <- ggplot(ens_all_up , aes(x=Count, y=reorder(ID, Count), fill=p.adjust)) + geom_bar(stat = "identity") + ggtitle("ENS Upregulated Pathways - GO:BP") + facet_wrap(~category) + theme_bw() + labels + axis + legend + scale_fill_continuous(type="viridis")
# gg_dn_ens <- ggplot(ens_all_dn, aes(x=Count, y=reorder(ID, Count), fill=p.adjust)) + geom_bar(stat = "identity") + ggtitle("ENS Downregulated Pathways - GO:BP") + facet_wrap(~category) + theme_bw() + labels + axis + legend + scale_fill_continuous(type="viridis")
#
#
# grid.arrange(gg_up_ens, gg_dn_ens, ncol=1, nrow = 2)

```

### Nippo

```{r, GO Plots 1Nippo, echo=FALSE, warning=FALSE, fig.height=6, fig.width=14, fig.fullwidth=TRUE}
grid.arrange(gg_up_ens_nippo, gg_dn_ens_nippo , ncol=2, nrow = 1)
# goplot(egobp_dn)

```

<br>

```{r GO Result nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


GO_ens_nippo <- rbind(GO_up_ens_nippo, GO_down_ens_nippo)

GO_ens_nippo %>%
  gt() %>%
  tab_header(
    title = 'ENS Nippo vs Naive',
    subtitle = 'clusterProfiler Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

<br><br>


### DSS

```{r, GO Plots 1DSS, echo=FALSE, warning=FALSE, fig.height=6, fig.width=14, fig.fullwidth=TRUE}
grid.arrange(gg_up_ens_dss, gg_dn_ens_dss , ncol=2, nrow = 1)
# goplot(egobp_dn)

```

<br>

```{r GO Result dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


GO_ens_dss <- rbind(GO_up_ens_dss, GO_down_ens_dss)

GO_ens_dss %>%
  gt() %>%
  tab_header(
    title = 'ENS DSS vs Naive',
    subtitle = 'clusterProfiler Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

<br><br>

### Th17

```{r, GO Plots 1Th17, echo=FALSE, warning=FALSE, fig.height=6, fig.width=14, fig.fullwidth=TRUE}
grid.arrange(gg_up_ens_th17, gg_dn_ens_th17 , ncol=2, nrow = 1)
# goplot(egobp_dn)

```

<br>

```{r GO Result th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


GO_ens_th17 <- rbind(GO_up_ens_th17, GO_down_ens_th17)

GO_ens_th17 %>%
  gt() %>%
  tab_header(
    title = 'ENS Th17 vs Naive',
    subtitle = 'clusterProfiler Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

<br> <br> <br>


```{r GSE with cluster Profiler KEGG WikiPathways Reactome, message = FALSE, echo=FALSE, warning=FALSE}

######### ENS ############

## NIPPO ##

# we want the log2 fold changes
original_gene_list <- ens.nippo$Log2FC
# name the vector
names(original_gene_list) <- ens.nippo$GeneID
# omit any NA values
gene_list<-na.omit(original_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)


# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=org.Mm.eg.db)
# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df = ens.nippo[ens.nippo$GeneID %in% dedup_ids$ENSEMBL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df$ENTREZID = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df$Log2FC

# Name vector with ENTREZ ids
names(kegg_gene_list) <- df$ENTREZID

# omit any NA values
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


kegg_organism = "mmu"

k_ens_nippo <- gseKEGG(geneList = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "fdr",
               keyType       = "ncbi-geneid")

k_ens_nippo@result$Description <- gsub(k_ens_nippo@result$Description, pattern="\\s-\\sMus.+$", replacement="")

kt_ens_nippo <- k_ens_nippo@result

kegg_ens_nippo <- dotplot(k_ens_nippo, showCategory = 10, title = "ENS Nippo - KEGG" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) + scale_color_gradient(low="#FFB500",high="#1400FF")


## WikiPaths Nippo

w_ens_nippo <- gseWP(kegg_gene_list,
               organism = "Mus musculus")


wp_ens_nippo<- dotplot(w_ens_nippo, showCategory = 10, title = "WikiPathways: ENS - Nippo" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
        scale_color_gradient(low="#72E89A",high="#1400FF")

## Reactome

r_ens_nippo <- gsePathway(kegg_gene_list,
                organism = "mouse",
                pvalueCutoff = 0.2,
                pAdjustMethod = "BH",
                verbose = FALSE)


re_ens_nippo <- dotplot(r_ens_nippo, showCategory = 10, title = "REACTOME: ENS - Nippo" , split=".sign") +
        facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
        scale_color_gradient(low="#72E89A",high="#FFB500")


## DSS ##

# we want the log2 fold change
original_gene_list <- ens.dss$Log2FC
# name the vector
names(original_gene_list) <- ens.dss$GeneID
# omit any NA values
gene_list<-na.omit(original_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)


# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=org.Mm.eg.db)
# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df = ens.dss[ens.dss$GeneID %in% dedup_ids$ENSEMBL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df$ENTREZID = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df$Log2FC

# Name vector with ENTREZ ids
names(kegg_gene_list) <- df$ENTREZID

# omit any NA values
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


kegg_organism = "mmu"


k_ens_dss <- gseKEGG(geneList     = kegg_gene_list,
                       organism     = kegg_organism,
                       nPerm        = 10000,
                       minGSSize    = 3,
                       maxGSSize    = 800,
                       pvalueCutoff = 0.05,
                       pAdjustMethod = "fdr",
                       keyType       = "ncbi-geneid")

k_ens_dss <- setReadable(k_ens_dss, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

k_ens_dss@result$Description <- gsub(k_ens_dss@result$Description, pattern="\\s-\\sMus.+$", replacement="")

kt_ens_dss <- k_ens_dss@result

kegg_ens_dss <- dotplot(k_ens_dss, showCategory = 10, title = "KEGG: ENS - DSS" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) + scale_color_gradient(low="#FFB500",high="#1400FF")

## WikiPaths Nippo

w_ens_dss <- gseWP(kegg_gene_list,
               organism = "Mus musculus")

w_ens_dss <- setReadable(w_ens_dss, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

wp_ens_dss <- dotplot(w_ens_dss, showCategory = 10, title = "WikiPathways: ENS - DSS" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
        scale_color_gradient(low="#72E89A",high="#1400FF")



## Reactome

r_ens_dss <- gsePathway(kegg_gene_list,
                organism = "mouse",
                pvalueCutoff = 0.2,
                pAdjustMethod = "BH",
                verbose = FALSE)

r_ens_dss <- setReadable(r_ens_dss, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

re_ens_dss <- dotplot(r_ens_dss, showCategory = 10, title = "REACTOME: ENS - DSS" , split=".sign") +
        facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
        scale_color_gradient(low="#72E89A",high="#FFB500")


## Th17 ##

# we want the log2 fold change
original_gene_list <- ens.th17$Log2FC
# name the vector
names(original_gene_list) <- ens.th17$GeneID
# omit any NA values
gene_list<-na.omit(original_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)


# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=org.Mm.eg.db)
# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df = ens.th17[ens.th17$GeneID %in% dedup_ids$ENSEMBL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df$ENTREZID = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df$Log2FC

# Name vector with ENTREZ ids
names(kegg_gene_list) <- df$ENTREZID

# omit any NA values
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


kegg_organism = "mmu"

k_ens_th17 <- gseKEGG(geneList     = kegg_gene_list,
                       organism     = kegg_organism,
                       nPerm        = 10000,
                       minGSSize    = 3,
                       maxGSSize    = 800,
                       pvalueCutoff = 0.05,
                       pAdjustMethod = "fdr",
                       keyType       = "ncbi-geneid")

k_ens_th17@result$Description <- gsub(k_ens_th17@result$Description, pattern="\\s-\\sMus.+$", replacement="")

kt_ens_th17 <- k_ens_th17@result

kegg_ens_th17 <- dotplot(k_ens_th17, showCategory = 10, title = "KEGG: ENS - Th17" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) + scale_color_gradient(low="#FFB500",high="#1400FF")

## WikiPaths Nippo

w_ens_th17 <- gseWP(kegg_gene_list,
               organism = "Mus musculus")

wp_ens_th17 <- dotplot(w_ens_th17, showCategory = 10, title = "WikiPathways: ENS - Th17" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
        scale_color_gradient(low="#72E89A",high="#1400FF")

## Reactome

r_ens_th17 <- gsePathway(kegg_gene_list,
                organism = "mouse",
                pvalueCutoff = 0.2,
                pAdjustMethod = "BH",
                verbose = FALSE)


re_ens_th17 <- dotplot(r_ens_th17, showCategory = 10, title = "REACTOME: ENS - Th17" , split=".sign") +
        facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
        scale_color_gradient(low="#72E89A",high="#FFB500")




```

## KEGG {.tabset}

```{r, GO Plots KEGG_ENS, echo=FALSE, warning=FALSE, fig.height=8, fig.width=22, fig.fullwidth=TRUE}
grid.arrange(kegg_ens_nippo, kegg_ens_dss , kegg_ens_th17, ncol=3, nrow = 1)
# goplot(egobp_dn)

```

### Nippo

```{r Kegg Result nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}



kt_ens_nippo %>%
  gt() %>%
  tab_header(
    title = 'ENS Nippo vs Naive',
    subtitle = 'KEGG Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### DSS

```{r Kegg Result dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


kt_ens_dss %>%
  gt() %>%
  tab_header(
    title = 'ENS DSS vs Naive',
    subtitle = 'KEGG Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```


### Th17

```{r Kegg Result Th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


kt_ens_th17 %>%
  gt() %>%
  tab_header(
    title = 'ENS Th17 vs Naive',
    subtitle = 'KEGG Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

<br><br><br>

## WikiPathways {.tabset}

```{r, GO Plots WP_ENS, echo=FALSE, warning=FALSE, fig.height=8, fig.width=22, fig.fullwidth=TRUE}
grid.arrange(wp_ens_nippo, wp_ens_dss , wp_ens_th17, ncol=3, nrow = 1)
# goplot(egobp_dn)

```

### Nippo

```{r Wikipaths Result nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

w_ens_nippo@result %>%
  gt() %>%
  tab_header(
    title = 'ENS Nippo vs Naive',
    subtitle = 'WikiPathways Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### DSS

```{r Wikipaths Result dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

w_ens_dss@result %>%
  gt() %>%
  tab_header(
    title = 'ENS DSS vs Naive',
    subtitle = 'WikiPathways Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### Th17

```{r Wikipaths Result Th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

w_ens_th17@result %>%
  gt() %>%
  tab_header(
    title = 'ENS Th17 vs Naive',
    subtitle = 'WikiPathways Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```
<br <br><br>

## Reactome {.tabset}

```{r, GO Plots Reactome_ENS, echo=FALSE, warning=FALSE, fig.height=8, fig.width=22, fig.fullwidth=TRUE}
grid.arrange(re_ens_nippo, re_ens_dss , re_ens_th17, ncol=3, nrow = 1)
# goplot(egobp_dn)

```

### Nippo

```{r Reactome Nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

r_ens_nippo@result %>%
  gt() %>%
  tab_header(
    title = 'ENS Nippo vs Naive',
    subtitle = 'Reactome Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### DSS

```{r Reactome DSS, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

r_ens_dss@result %>%
  gt() %>%
  tab_header(
    title = 'ENS DSS vs Naive',
    subtitle = 'Reactome Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### Th17

```{r Reactome Th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

r_ens_th17@result %>%
  gt() %>%
  tab_header(
    title = 'ENS Th17 vs Naive',
    subtitle = 'Reactome Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```
