---
title: "VG in inflammation: DGE & GSEA"
author: "Patrycja Monika Topczewska"
date: "`r Sys.Date()`"
output:
      html_document:
        toc: true
        toc_float: true
        number_sections: true
        collapsed: true
        fig_caption: yes
        df_print: paged
---

DGE and Gene Ontology Analysis of the Vagal Ganglion during Inflammation

VG of infected and naive Snap25CreINTACT mice were isolated. Nuclei isolation was performed and neuronal nuclei were sort-purified (GFP+). Bulk RNA-Sequencing was done and the results are shownbelow.

```{r setup, include=FALSE, message = FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Volumes/PortableSSD/1_PhD/1_Projects/1_Neurons/2_SeqAnalysis/1_Mine")

# load importante libraries

library(DESeq2)
library(limma)
library(gProfileR)
library(httr)
library(RJSONIO)
library(pheatmap)
library(dbplyr)
library(dplyr)
library(ggplot2)
library(tidyverse)
library(hrbrthemes)
library(viridis)
library(plotly)
library(RColorBrewer)
library(viridis)
library(readxl)
#library(DEGreport)
library(htmltools)
library(vctrs)
library(EnhancedVolcano)
library(readr)
library(topGO)
library(org.Mm.eg.db)
library(AnnotationDbi)
library(DOSE)
library(GO.db)
library(GSEABase)
library(clusterProfiler)
library(tidyr)
library(stringr)
library(rWikiPathways)
library(RCy3)
library(ReactomePA)
library(scales)
library(gridExtra)
library(grid)
library(ggrepel)
library(repr)
library(ggpubr)
library(knitr)
library(kableExtra)
library(DT)
library(xfun)
library(rmarkdown)
library(patchwork)
library(readr)
library(gprofiler2)
library(ggplot2)
library(gt)
library(gtExtras)
```

```{r Import data, echo=FALSE,warning=FALSE,message = FALSE}


#### norm counts

Neurons_NormCounts_symbol <- read_csv("../data/processed/Neurons_NormCounts_symbol.csv")
nc <- as.data.frame(Neurons_NormCounts_symbol)
nc <- unique(nc)
nc <- nc[!duplicated(nc$gene_name),]
nc <- na.omit(nc)
rownames(nc) <- nc$gene_name


### DESeq2



## for Volcano

vg.nippo.v <- read.delim("../data/raw/DESeq2_VG_Nippo_results.tabular", header=F)
vg.dss.v <- read.delim("../data/raw/DESeq2_VG_DSS_results.tabular", header=F)
vg.th17.v <- read.delim("../data/raw/DESeq2_VG_Th17_results.tabular", header=F)



colnames(vg.nippo.v) <- c("GeneID","BaseMean","Log2FC","StdErr","WaldStat","pval","padj")
colnames(vg.dss.v) <- c("GeneID","BaseMean","Log2FC","StdErr","WaldStat","pval","padj")
colnames(vg.th17.v) <- c("GeneID","BaseMean","Log2FC","StdErr","WaldStat","pval","padj")




anno_df <- data.frame(GeneID = vg.nippo.v$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
vg.nippo.v <- merge(vg.nippo.v, anno_df, by = "GeneID", all=TRUE)

anno_df <- data.frame(GeneID = vg.dss.v$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
vg.dss.v <- merge(vg.dss.v, anno_df, by = "GeneID", all=TRUE)

anno_df <- data.frame(GeneID = vg.th17.v$GeneID, stringsAsFactors = FALSE)
anno_df$GeneName <- mapIds(org.Mm.eg.db,
                            keys = anno_df$GeneID,
                            column = "SYMBOL",
                            keytype = "ENSEMBL",
                            multiVals = "first")
vg.th17.v <- merge(vg.th17.v, anno_df, by = "GeneID", all=TRUE)



vg.nippo <- vg.nippo.v
vg.nippo <- vg.nippo[vg.nippo$BaseMean > 30, ]
vg.nippo <- vg.nippo[vg.nippo$padj <= 0.05, ]
vg.nippo <- na.omit(vg.nippo)

vg.dss <- vg.dss.v
vg.dss <- vg.dss[vg.dss$BaseMean > 30, ]
vg.dss <- vg.dss[vg.dss$padj <= 0.05, ]
vg.dss <- na.omit(vg.dss)

vg.th17 <- vg.th17.v
vg.th17 <- vg.th17[vg.th17$BaseMean > 30, ]
vg.th17 <- vg.th17[vg.th17$padj <= 0.05, ]
vg.th17 <- na.omit(vg.th17)


```

# DESeq2

## Results {.tabset}
### Nippo
```{r DGESeq2 Results Nippo, echo=FALSE, paged.print=TRUE, results='asis',warning=FALSE}



vg.nippo1 <- transform(vg.nippo, pvalue=sapply(pval, formatC, format = "e", digits=3))
vg.nippo1 <- transform(vg.nippo1, padj=sapply(padj, formatC, format = "e", digits=3))
vg.nippo1 <-vg.nippo1  %>% mutate_at(vars("BaseMean", "Log2FC", "StdErr", "WaldStat"), funs(round(., 3)))


vg.nippo1 %>%
  gt() %>%
  tab_header(
    title = 'DGE: VG Nippo vs Naive',
    subtitle = 'DESeq2 Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### DSS

```{r DGESeq2 Results DSS, echo=FALSE, paged.print=TRUE, results='asis',warning=FALSE}



vg.dss1 <- transform(vg.dss, pvalue=sapply(pval, formatC, format = "e", digits=3))
vg.dss1 <- transform(vg.dss1, padj=sapply(padj, formatC, format = "e", digits=3))
vg.dss1 <-vg.dss1  %>% mutate_at(vars("BaseMean", "Log2FC", "StdErr", "WaldStat"), funs(round(., 3)))


vg.dss1 %>%
  gt() %>%
  tab_header(
    title = 'DGE: VG DSS vs Naive',
    subtitle = 'DESeq2 Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```

### Th17

```{r DGESeq2 Results th17, echo=FALSE, paged.print=TRUE, results='asis',warning=FALSE}



vg.th171 <- transform(vg.th17, pvalue=sapply(pval, formatC, format = "e", digits=3))
vg.th171 <- transform(vg.th171, padj=sapply(padj, formatC, format = "e", digits=3))
vg.th171 <-vg.th171  %>% mutate_at(vars("BaseMean", "Log2FC", "StdErr", "WaldStat"), funs(round(., 3)))


vg.th171 %>%
  gt() %>%
  tab_header(
    title = 'DGE: VG Th17 vs Naive',
    subtitle = 'DESeq2 Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  )

```


<br>

## Volcano Plots {.tabset}

```{r Volcano Plots, echo=FALSE, warning=FALSE, message = FALSE}

######## nippo

vg.nippo.v <- na.omit(vg.nippo.v)

top_up <- vg.nippo.v[vg.nippo.v$Log2FC > 1.5 & vg.nippo.v$padj < 0.05,]
top_up <- top_up[order(-top_up$Log2FC),]

top_down <- vg.nippo.v[vg.nippo.v$Log2FC < -1.5 & vg.nippo.v$padj < 0.05,]
top_down <- top_down[order(top_down$Log2FC),]


top <- c(top_up[1:50,"GeneName"],top_down[1:50,"GeneName"])


V.nippo <- EnhancedVolcano(vg.nippo.v,
                      lab = vg.nippo.v$GeneName,
                      selectLab = c(top),
                      #boxedLabels = TRUE,
                      #parseLabels = TRUE,
                      drawConnectors = TRUE,
                      widthConnectors = 0.25,
                      colConnectors = c("black"),
                      x = 'Log2FC',
                      y = 'pval',
                      title = 'VG: Nippo vs Naive',
                      xlim = c(-6,6),
                      ylim = c(0,20),
                      pCutoff = 0.05,
                      FCcutoff = 0.5,
                      pointSize = 1.5,
                      labSize = 3.5,
                      col = c('#3e3e40', '#3e3e40', '#3e3e40',"#0066a1"),
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)


######## DSS

vg.dss.v <- na.omit(vg.dss.v)

top_up <- vg.dss.v[vg.dss.v$Log2FC > 1.5 & vg.dss.v$padj < 0.05,]
top_up <- top_up[order(-top_up$Log2FC),]

top_down <- vg.dss.v[vg.dss.v$Log2FC < -1.5 & vg.dss.v$padj < 0.05,]
top_down <- top_down[order(top_down$Log2FC),]


top <- c(top_up[1:50,"GeneName"],top_down[1:50,"GeneName"])


V.dss <- EnhancedVolcano(vg.dss.v,
                      lab = vg.dss.v$GeneName,
                      selectLab = c(top),
                      #boxedLabels = TRUE,
                      #parseLabels = TRUE,
                      drawConnectors = TRUE,
                      widthConnectors = 0.25,
                      colConnectors = c("black"),
                      x = 'Log2FC',
                      y = 'pval',
                      title = 'VG: DSS vs Naive',
                      xlim = c(-6,6),
                      ylim = c(0,20),
                      pCutoff = 0.05,
                      FCcutoff = 0.5,
                      pointSize = 1.5,
                      labSize = 3.5,
                      col = c('#3e3e40', '#3e3e40', '#3e3e40',"red"),
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)


######## Th17

vg.th17.v <- na.omit(vg.th17.v)

top_up <- vg.th17.v[vg.th17.v$Log2FC > 1.5 & vg.th17.v$padj < 0.05,]
top_up <- top_up[order(-top_up$Log2FC),]

top_down <- vg.th17.v[vg.th17.v$Log2FC < -1.5 & vg.th17.v$padj < 0.05,]
top_down <- top_down[order(top_down$Log2FC),]


top <- c(top_up[1:50,"GeneName"],top_down[1:50,"GeneName"])


V.th17 <- EnhancedVolcano(vg.th17.v,
                      lab = vg.th17.v$GeneName,
                      selectLab = c(top),
                      #boxedLabels = TRUE,
                      #parseLabels = TRUE,
                      drawConnectors = TRUE,
                      widthConnectors = 0.25,
                      colConnectors = c("black"),
                      x = 'Log2FC',
                      y = 'pval',
                      title = 'VG: Th17 vs Naive',
                      xlim = c(-6,6),
                      ylim = c(0,20),
                      pCutoff = 0.05,
                      FCcutoff = 0.5,
                      pointSize = 1.5,
                      labSize = 3.5,
                      col = c('#3e3e40', '#3e3e40', '#3e3e40',"darkgreen"),
                      gridlines.major = FALSE,
                      gridlines.minor = FALSE)


```

```{r enhancedVolcano plot, message = FALSE, echo=FALSE, warning=FALSE, fig.height=12,fig.width=25, fig.fullwidth=TRUE}

ggarrange(V.nippo, V.dss, V.th17, ncol=3, nrow=1)

```
<br><br><br><br>

Gene Lists are filtered for Log2FoldChange: upregulated \> 1.5 & downregulated \< -1.5 and padj \< 0.05 For Analysis, gene symbols were used, and ordered query was performed

# GSEA

## gProfiler Enrichment Analysis {.tabset}

```{r GSEA with  gProfiler, message = FALSE, echo=FALSE, warning=FALSE, results='hide'}

## VG Nippo



up.genes <- vg.nippo[vg.nippo$Log2FC > 1.5, ]
up.genes.ordered <- up.genes[order(up.genes$Log2FC, decreasing = TRUE),]

dn.genes <- vg.nippo[vg.nippo$Log2FC < -1.5, ]
dn.genes.ordered <- dn.genes[order(up.genes$Log2FC, decreasing = FALSE),]

multi_gp_vg_nippo = gost(list("up-regulated" = up.genes.ordered$GeneName,
                     "down-regulated" = dn.genes.ordered$GeneName),organism = "mmusculus", ordered_query = TRUE)

gp_vg_nippo <- gostplot(multi_gp_vg_nippo, interactive = TRUE)

## DSS




up.genes <- vg.dss[vg.dss$Log2FC > 1.5, ]
up.genes.ordered <- up.genes[order(up.genes$Log2FC, decreasing = TRUE),]

dn.genes <- vg.dss[vg.dss$Log2FC < -1.5, ]
dn.genes.ordered <- dn.genes[order(up.genes$Log2FC, decreasing = FALSE),]


multi_gp_vg_dss = gost(list("up-regulated" = up.genes.ordered$GeneName,
                     "down-regulated" = dn.genes.ordered$GeneName),organism = "mmusculus", ordered_query = TRUE)



gp_vg_dss <- gostplot(multi_gp_vg_dss, interactive = TRUE)

## Th17



up.genes <- vg.th17[vg.th17$Log2FC > 1.5, ]
up.genes.ordered <- up.genes[order(up.genes$Log2FC, decreasing = TRUE),]

dn.genes <- vg.th17[vg.th17$Log2FC < -1.5, ]
dn.genes.ordered <- dn.genes[order(up.genes$Log2FC, decreasing = FALSE),]


multi_gp_vg_th17 = gost(list("up-regulated" = up.genes.ordered$GeneName,
                     "down-regulated" = dn.genes.ordered$GeneName),organism = "mmusculus", ordered_query = TRUE)



gp_vg_th17 <- gostplot(multi_gp_vg_th17, interactive = TRUE)


```

### Nippo

```{r gProfiler nippo plot, message = FALSE, echo=FALSE, warning=FALSE}

ggplotly(gp_vg_nippo)

```

<br>

```{r gProfiler nippo, message = FALSE, echo=FALSE, warning=FALSE}


gp_vg_nippo_table <- multi_gp_vg_nippo[["result"]]
gp_vg_nippo_table <- transform(gp_vg_nippo_table, p_value=sapply(p_value, formatC, format = "e", digits=3))
gp_vg_nippo_table <- gp_vg_nippo_table %>% mutate_at(vars("precision","recall"), funs(round(., 3)))
gp_vg_nippo_table <- subset(gp_vg_nippo_table, select=-c(2))
table_vg_nippo <- gp_vg_nippo_table[,c("term_id","term_name","source","query","p_value","term_size","query_size","intersection_size","precision","effective_domain_size","parents")]

table_vg_nippo %>%
  gt() %>%
  tab_header(
    title = 'VG Nippo vs Naive',
    subtitle = 'gProfiler Analysis'
  ) %>%
  cols_label(
   term_id = "Term ID",
   term_name = "Term Name",
   source = "Source",
   query = "Up/Down",
   p_value = "P-Value",
   term_size = "Term Size",
   query_size = "Query Size",
   intersection_size = "Intersection Size",
   precision = "Precision",
   effective_domain_size = "Effective Domain Size",
   parents = "Parents"
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    term_name ~ px(350)
  )



```

<br><br>

### DSS

```{r gProfiler dss plot, message = FALSE, echo=FALSE, warning=FALSE}

ggplotly(gp_vg_dss)

```

<br>

```{r gProfiler dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

gp_vg_dss_table <- multi_gp_vg_dss[["result"]]
gp_vg_dss_table <- transform(gp_vg_dss_table, p_value=sapply(p_value, formatC, format = "e", digits=3))
gp_vg_dss_table <- gp_vg_dss_table %>% mutate_at(vars("precision","recall"), funs(round(., 3)))
gp_vg_dss_table <- subset(gp_vg_dss_table, select=-c(2))
table_vg_dss <- gp_vg_dss_table[,c("term_id","term_name","source","query","p_value","term_size","query_size","intersection_size","precision","effective_domain_size","parents")]

table_vg_dss %>%
  gt() %>%
  tab_header(
    title = 'VG DSS vs Naive',
    subtitle = 'gProfiler Analysis'
  ) %>%
  cols_label(
   term_id = "Term ID",
   term_name = "Term Name",
   source = "Source",
   query = "Up/Down",
   p_value = "P-Value",
   term_size = "Term Size",
   query_size = "Query Size",
   intersection_size = "Intersection Size",
   precision = "Precision",
   effective_domain_size = "Effective Domain Size",
   parents = "Parents"
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    term_name ~ px(350)
  )


```

<br><br>


### Th17

```{r gProfiler Th17 plot, message = FALSE, echo=FALSE, warning=FALSE}

ggplotly(gp_vg_th17)

```

<br>

```{r Result th17 , echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

gp_vg_th17_table <- multi_gp_vg_th17[["result"]]
gp_vg_th17_table <- transform(gp_vg_th17_table, p_value=sapply(p_value, formatC, format = "e", digits=3))
gp_vg_th17_table <- gp_vg_th17_table %>% mutate_at(vars("precision","recall"), funs(round(., 3)))
gp_vg_th17_table <- subset(gp_vg_th17_table, select=-c(2))
table_vg_th17 <- gp_vg_th17_table[,c("term_id","term_name","source","query","p_value","term_size","query_size","intersection_size","precision","effective_domain_size","parents")]

table_vg_th17 %>%
  gt() %>%
  tab_header(
    title = 'VG Th17 vs Naive',
    subtitle = 'gProfiler Analysis'
  ) %>%
  cols_label(
   term_id = "Term ID",
   term_name = "Term Name",
   source = "Source",
   query = "Up/Down",
   p_value = "P-Value",
   term_size = "Term Size",
   query_size = "Query Size",
   intersection_size = "Intersection Size",
   precision = "Precision",
   effective_domain_size = "Effective Domain Size",
   parents = "Parents"
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    term_name ~ px(350)
  )



```


<br> <br> <br>

## clusterProfiler Enrichment Analysis {.tabset}

```{r GSEA with cluster Profiler, message = FALSE, echo=FALSE, warning=FALSE, results='hide'}

###### VG #################################



### NIPPO ###

## gene list for upregulated genes
up.genes <- vg.nippo[vg.nippo$Log2FC > 1.5, ]
up.genes.entrez <- bitr(up.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## gene list for downregulated genes
dn.genes <- vg.nippo[vg.nippo$Log2FC < -1.5, ]
dn.genes.entrez <- bitr(dn.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## backbone gene list (all genes)
bkgd.genes <- Neurons_NormCounts_symbol$GeneID
bkgd.genes.entrez <- bitr(bkgd.genes,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


##  https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/  #####

ego_up_vg_nippo <- clusterProfiler::enrichGO(
  gene     = up.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  minGSSize = 3,
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


ego_dn_vg_nippo <- clusterProfiler::enrichGO(
  gene     = dn.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  minGSSize = 3,
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)

GO_up_vg_nippo <- as.data.frame(ego_up_vg_nippo)
GO_up_vg_nippo$query <- c("up-regulated")

GO_down_vg_nippo <- as.data.frame(ego_dn_vg_nippo)
#GO_down_vg_nippo$query <- c("down-regulated")



### DSS ###

## gene list for upregulated genes
up.genes <- vg.dss[vg.dss$Log2FC > 1.5, ]
up.genes.entrez <- bitr(up.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## gene list for downregulated genes
dn.genes <- vg.dss[vg.dss$Log2FC < -1.5, ]
dn.genes.entrez <- bitr(dn.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


##  https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/  #####

ego_up_vg_dss <- clusterProfiler::enrichGO(
  gene     = up.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  minGSSize = 3,
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)



ego_dn_vg_dss <- clusterProfiler::enrichGO(
  gene     = dn.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  minGSSize = 3,
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


GO_up_vg_dss <- as.data.frame(ego_up_vg_dss)
GO_up_vg_dss$query <- c("up-regulated")

GO_down_vg_dss <- as.data.frame(ego_dn_vg_dss)
GO_down_vg_dss$query <- c("down-regulated")

### Th17 ###

## gene list for upregulated genes
up.genes <- vg.th17[vg.th17$Log2FC > 1.5, ]
up.genes.entrez <- bitr(up.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)
## gene list for downregulated genes
dn.genes <- vg.th17[vg.th17$Log2FC < -1.5, ]
dn.genes.entrez <- bitr(dn.genes$GeneID,fromType = "ENSEMBL",toType = "ENTREZID",OrgDb = org.Mm.eg.db)


##  https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/  #####

ego_up_vg_th17 <- clusterProfiler::enrichGO(
  gene     = up.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  minGSSize = 3,
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)



ego_dn_vg_th17 <- clusterProfiler::enrichGO(
  gene     = dn.genes.entrez[[2]],
  universe = bkgd.genes.entrez[[2]],
  OrgDb    = org.Mm.eg.db,
  ont      = "ALL",
  minGSSize = 3,
  pAdjustMethod = "fdr",
  pvalueCutoff = 0.05, #p.adjust cutoff (https://github.com/GuangchuangYu/clusterProfiler/issues/104)
  readable = TRUE)


GO_up_vg_th17 <- as.data.frame(ego_up_vg_th17)
GO_up_vg_th17$query <- c("up-regulated")

GO_down_vg_th17 <- as.data.frame(ego_dn_vg_th17)
GO_down_vg_th17$query <- c("down-regulated")

```

```{r GSEA with cluster Profiler enrichGO, message = FALSE, echo=FALSE, warning=FALSE}


gg_up_vg_nippo <- dotplot(ego_up_vg_nippo, showCategory=10) + ggtitle("VG Nippo - UP") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))
#gg_dn_vg_nippo <- dotplot(ego_dn_vg_nippo, showCategory=10) + ggtitle("VG Nippo - DOWN") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))

gg_up_vg_dss <- dotplot(ego_up_vg_dss, showCategory=10) + ggtitle("VG DSS - UP") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))
gg_dn_vg_dss <- dotplot(ego_dn_vg_dss, showCategory=10) + ggtitle("VG DSS - DOWN") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))

gg_up_vg_th17 <- dotplot(ego_up_vg_th17, showCategory=10) + ggtitle("VG Th17 - UP") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))
gg_dn_vg_th17 <- dotplot(ego_dn_vg_th17, showCategory=10) + ggtitle("VG Th17 - DOWN") + scale_y_discrete(labels = function(x) str_wrap(x, width = 50))

# legend <-  theme(legend.key.size = unit(0.2, 'cm'), #change legend key size
#                  legend.key.height = unit(0.5, 'cm'), #change legend key height
#                  legend.key.width = unit(0.3, 'cm'), #change legend key width
#                  legend.title = element_text(size=8), #change legend title font size
#                  legend.text = element_text(size=8)) #change legend text font size
#
# labels <- scale_y_discrete(labels = function(x) str_wrap(x, width = 200))
# axis <-  theme(axis.text.y=element_text(size=10),axis.text.x=element_text(size=10), axis.title.x=element_text(size=8),axis.title.y=element_text(size=0),legend.text = element_text(size = 2))
#
#
# ### make barplots from GO_BP top 10
#
# gg_up_vg <- ggplot(vg_all_up , aes(x=Count, y=reorder(ID, Count), fill=p.adjust)) + geom_bar(stat = "identity") + ggtitle("VG Upregulated Pathways - GO:BP") + facet_wrap(~category) + theme_bw() + labels + axis + legend + scale_fill_continuous(type="viridis")
# gg_dn_vg <- ggplot(vg_all_dn, aes(x=Count, y=reorder(ID, Count), fill=p.adjust)) + geom_bar(stat = "identity") + ggtitle("VG Downregulated Pathways - GO:BP") + facet_wrap(~category) + theme_bw() + labels + axis + legend + scale_fill_continuous(type="viridis")
#
#
# grid.arrange(gg_up_vg, gg_dn_vg, ncol=1, nrow = 2)

```

### Nippo

```{r, GO Plots 1Nippo, echo=FALSE, warning=FALSE, fig.height=6, fig.width=14, fig.fullwidth=TRUE}
grid.arrange(gg_up_vg_nippo, ncol=2, nrow = 1)
# goplot(egobp_dn)

```

<br>

```{r GO Result nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


GO_vg_nippo <- rbind(GO_up_vg_nippo, GO_down_vg_nippo)

GO_vg_nippo %>%
  gt() %>%
  tab_header(
    title = 'VG Nippo vs Naive',
    subtitle = 'clusterProfiler Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(350)
  )

```

<br><br>


### DSS

```{r, GO Plots 1DSS, echo=FALSE, warning=FALSE, fig.height=6, fig.width=14, fig.fullwidth=TRUE}
grid.arrange(gg_up_vg_dss, gg_dn_vg_dss , ncol=2, nrow = 1)
# goplot(egobp_dn)

```

<br>

```{r GO Result dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


GO_vg_dss <- rbind(GO_up_vg_dss, GO_down_vg_dss)

GO_vg_dss %>%
  gt() %>%
  tab_header(
    title = 'VG DSS vs Naive',
    subtitle = 'clusterProfiler Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(350)
  )

```

<br><br>

### Th17

```{r, GO Plots 1Th17, echo=FALSE, warning=FALSE, fig.height=6, fig.width=14, fig.fullwidth=TRUE}
grid.arrange(gg_up_vg_th17, gg_dn_vg_th17 , ncol=2, nrow = 1)
# goplot(egobp_dn)

```

<br>

```{r GO Result th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


GO_vg_th17 <- rbind(GO_up_vg_th17, GO_down_vg_th17)

GO_vg_th17 %>%
  gt() %>%
  tab_header(
    title = 'VG Th17 vs Naive',
    subtitle = 'clusterProfiler Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(350)
  )

```

<br> <br> <br>


```{r GSE with cluster Profiler KEGG WikiPathways Reactome, message = FALSE, echo=FALSE, warning=FALSE}

######### VG ############



## NIPPO ##

# we want the log2 fold changes
original_gene_list <- vg.nippo$Log2FC
# name the vector
names(original_gene_list) <- vg.nippo$GeneID
# omit any NA values
gene_list<-na.omit(original_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)


# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=org.Mm.eg.db)
# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df = vg.nippo[vg.nippo$GeneID %in% dedup_ids$ENSEMBL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df$ENTREZID = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df$Log2FC

# Name vector with ENTREZ ids
names(kegg_gene_list) <- df$ENTREZID

# omit any NA values
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


kegg_organism = "mmu"

k_vg_nippo <- gseKEGG(geneList = kegg_gene_list,
               organism     = kegg_organism,
               nPerm        = 10000,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "fdr",
               keyType       = "ncbi-geneid")

k_vg_nippo <- setReadable(k_vg_nippo, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

k_vg_nippo@result$Description <- gsub(k_vg_nippo@result$Description, pattern="\\s-\\sMus.+$", replacement="")

kt_vg_nippo <- k_vg_nippo@result

kegg_vg_nippo <- dotplot(k_vg_nippo, showCategory = 10, title = "VG Nippo - KEGG" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) + scale_color_gradient(low="#FFB500",high="#1400FF")


## WikiPaths Nippo

w_vg_nippo <- gseWP(kegg_gene_list,
               organism = "Mus musculus")

w_vg_nippo <- setReadable(w_vg_nippo, OrgDb = org.Mm.eg.db, keyType="ENTREZID")
#wp_vg_nippo<- dotplot(w_vg_nippo, showCategory = 10, title = "WikiPathways: VG - Nippo" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
#        scale_color_gradient(low="#72E89A",high="#1400FF")

## Reactome

r_vg_nippo <- gsePathway(kegg_gene_list,
                organism = "mouse",
                pvalueCutoff = 0.2,
                pAdjustMethod = "BH",
                verbose = FALSE)

r_vg_nippo <- setReadable(r_vg_nippo, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

re_vg_nippo <- dotplot(r_vg_nippo, showCategory = 10, title = "REACTOME: VG - Nippo" , split=".sign") +
        facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
        scale_color_gradient(low="#72E89A",high="#FFB500")


## DSS ##

# we want the log2 fold change
original_gene_list <- vg.dss$Log2FC
# name the vector
names(original_gene_list) <- vg.dss$GeneID
# omit any NA values
gene_list<-na.omit(original_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)


# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=org.Mm.eg.db)
# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df = vg.dss[vg.dss$GeneID %in% dedup_ids$ENSEMBL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df$ENTREZID = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df$Log2FC

# Name vector with ENTREZ ids
names(kegg_gene_list) <- df$ENTREZID

# omit any NA values
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


kegg_organism = "mmu"

k_vg_dss <- gseKEGG(geneList     = kegg_gene_list,
                       organism     = kegg_organism,
                       nPerm        = 10000,
                       minGSSize    = 3,
                       maxGSSize    = 800,
                       pvalueCutoff = 0.05,
                       pAdjustMethod = "fdr",
                       keyType       = "ncbi-geneid")

k_vg_dss <- setReadable(k_vg_dss, OrgDb = org.Mm.eg.db, keyType="ENTREZID")


k_vg_dss@result$Description <- gsub(k_vg_dss@result$Description, pattern="\\s-\\sMus.+$", replacement="")

kt_vg_dss <- k_vg_dss@result

kegg_vg_dss <- dotplot(k_vg_dss, showCategory = 10, title = "KEGG: VG - DSS" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) + scale_color_gradient(low="#FFB500",high="#1400FF")

## WikiPaths Nippo

w_vg_dss <- gseWP(kegg_gene_list,
               organism = "Mus musculus")

w_vg_dss <- setReadable(w_vg_dss, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

wp_vg_dss <- dotplot(w_vg_dss, showCategory = 10, title = "WikiPathways: VG - DSS" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
        scale_color_gradient(low="#72E89A",high="#1400FF")


## Reactome

r_vg_dss <- gsePathway(kegg_gene_list,
                organism = "mouse",
                pvalueCutoff = 0.2,
                pAdjustMethod = "BH",
                verbose = FALSE)

r_vg_dss <- setReadable(r_vg_dss, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

re_vg_dss <- dotplot(r_vg_dss, showCategory = 10, title = "REACTOME: VG - DSS" , split=".sign") +
        facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
        scale_color_gradient(low="#72E89A",high="#FFB500")


## Th17 ##

# we want the log2 fold change
original_gene_list <- vg.th17$Log2FC
# name the vector
names(original_gene_list) <- vg.th17$GeneID
# omit any NA values
gene_list<-na.omit(original_gene_list)
# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)


# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(original_gene_list), fromType = "ENSEMBL", toType = "ENTREZID", OrgDb=org.Mm.eg.db)
# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("ENSEMBL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df = vg.th17[vg.th17$GeneID %in% dedup_ids$ENSEMBL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df$ENTREZID = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df$Log2FC

# Name vector with ENTREZ ids
names(kegg_gene_list) <- df$ENTREZID

# omit any NA values
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)


kegg_organism = "mmu"

k_vg_th17 <- gseKEGG(geneList     = kegg_gene_list,
                       organism     = kegg_organism,
                       nPerm        = 10000,
                       minGSSize    = 3,
                       maxGSSize    = 800,
                       pvalueCutoff = 0.05,
                       pAdjustMethod = "fdr",
                       keyType       = "ncbi-geneid")

k_vg_th17 <- setReadable(k_vg_th17, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

k_vg_th17@result$Description <- gsub(k_vg_th17@result$Description, pattern="\\s-\\sMus.+$", replacement="")

kt_vg_th17 <- k_vg_th17@result

kegg_vg_th17 <- dotplot(k_vg_th17, showCategory = 10, title = "KEGG: VG - Th17" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) + scale_color_gradient(low="#FFB500",high="#1400FF")

## WikiPaths Nippo

w_vg_th17 <- gseWP(kegg_gene_list,
               organism = "Mus musculus")

w_vg_th17 <- setReadable(w_vg_th17, OrgDb = org.Mm.eg.db, keyType="ENTREZID")

wp_vg_th17 <- dotplot(w_vg_th17, showCategory = 10, title = "WikiPathways: VG - Th17" , split=".sign") + facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 30)) +
        scale_color_gradient(low="#72E89A",high="#1400FF")

## Reactome

r_vg_th17 <- gsePathway(kegg_gene_list,
                organism = "mouse",
                pvalueCutoff = 0.2,
                pAdjustMethod = "BH",
                verbose = FALSE)

r_vg_th17 <- setReadable(r_vg_th17, OrgDb = org.Mm.eg.db, keyType="ENTREZID")


re_vg_th17 <- dotplot(r_vg_th17, showCategory = 10, title = "REACTOME: VG - Th17" , split=".sign") +
        facet_grid(.~.sign) + scale_y_discrete(labels = function(x) str_wrap(x, width = 50)) +
        scale_colour_gradient(low="#72E89A",high="#FFB500")




```

## KEGG {.tabset}

```{r, GO Plots KEGG_VG, echo=FALSE, warning=FALSE, fig.height=8, fig.width=22, fig.fullwidth=TRUE}
grid.arrange(kegg_vg_nippo, kegg_vg_dss , kegg_vg_th17, ncol=3, nrow = 1)
# goplot(egobp_dn)

```

### Nippo

```{r Kegg Result nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}



kt_vg_nippo %>%
  gt() %>%
  tab_header(
    title = 'VG Nippo vs Naive',
    subtitle = 'KEGG Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )

```

### DSS

```{r Kegg Result dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


kt_vg_dss %>%
  gt() %>%
  tab_header(
    title = 'VG DSS vs Naive',
    subtitle = 'KEGG Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )


```


### Th17

```{r Kegg Result Th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}


kt_vg_th17 %>%
  gt() %>%
  tab_header(
    title = 'VG Th17 vs Naive',
    subtitle = 'KEGG Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )


```

<br><br><br>

## WikiPathways {.tabset}

```{r, GO Plots WP_VG, echo=FALSE, warning=FALSE, fig.height=8, fig.width=22, fig.fullwidth=TRUE}
grid.arrange( wp_vg_dss , wp_vg_th17, ncol=3, nrow = 1)
# goplot(egobp_dn)

```

### Nippo

```{r Wikipaths Result nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

# w_vg_nippo@result %>%
#   gt() %>%
#   tab_header(
#     title = 'VG Nippo vs Naive',
#     subtitle = 'WikiPathways Analysis'
#   ) %>%
#   tab_options(
#     # These were the ones we applied in the first chapter
#     data_row.padding = px(2),
#     summary_row.padding = px(3), # A bit more padding for summaries
#     row_group.padding = px(4)    # And even more for our groups
#   ) %>%
#   opt_interactive(
#     use_search = TRUE,
#     use_filters = TRUE,
#     use_resizers = TRUE,
#     use_highlight = TRUE,
#     use_compact_mode = TRUE,
#     use_text_wrapping = FALSE,
#     use_page_size_select = TRUE
#   ) %>%
#   cols_width(
#     Description ~ px(300)
#   )


```

### DSS

```{r Wikipaths Result dss, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

w_vg_dss@result %>%
  gt() %>%
  tab_header(
    title = 'VG DSS vs Naive',
    subtitle = 'WikiPathways Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )

```

### Th17

```{r Wikipaths Result Th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

w_vg_th17@result %>%
  gt() %>%
  tab_header(
    title = 'VG Th17 vs Naive',
    subtitle = 'WikiPathways Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )


```
<br <br><br>

## Reactome {.tabset}

```{r, GO Plots Reactome_VG, echo=FALSE, warning=FALSE, fig.height=8, fig.width=22, fig.fullwidth=TRUE}
grid.arrange(re_vg_nippo, re_vg_dss , re_vg_th17, ncol=3, nrow = 1)
# goplot(egobp_dn)

```

### Nippo

```{r Reactome Nippo, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

r_vg_nippo@result %>%
  gt() %>%
  tab_header(
    title = 'VG Nippo vs Naive',
    subtitle = 'Reactome Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )


```

### DSS

```{r Reactome DSS, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

r_vg_dss@result %>%
  gt() %>%
  tab_header(
    title = 'VG DSS vs Naive',
    subtitle = 'Reactome Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )


```

### Th17

```{r Reactome Th17, echo=FALSE, paged.print=TRUE, results='asis',fig.height=5, fig.length=20,warning=FALSE}

r_vg_th17@result %>%
  gt() %>%
  tab_header(
    title = 'VG Th17 vs Naive',
    subtitle = 'Reactome Analysis'
  ) %>%
  tab_options(
    # These were the ones we applied in the first chapter
    data_row.padding = px(2),
    summary_row.padding = px(3), # A bit more padding for summaries
    row_group.padding = px(4)    # And even more for our groups
  ) %>%
  opt_interactive(
    use_search = TRUE,
    use_filters = TRUE,
    use_resizers = TRUE,
    use_highlight = TRUE,
    use_compact_mode = TRUE,
    use_text_wrapping = FALSE,
    use_page_size_select = TRUE
  ) %>%
  cols_width(
    Description ~ px(300)
  )


```
